cobbler/items/item.py

```diff
@@ -652,40 +834,97 @@ def parent(self, parent: str):
 
         :param parent: The new parent object. This needs to be a descendant in the logical inheritance chain.
         """
+        if not isinstance(parent, str):
+            raise TypeError('Property "parent" must be of type str!')
+        if not parent:
+            self._parent = ""
+            return
+        if parent == self.name:
+            # check must be done in two places as setting parent could be called before/after setting name...
+            raise CX("self parentage is weird")
+        found = self.api.get_items(self.COLLECTION_TYPE).get(parent)
+        if found is None:
+            raise CX(f'profile "{parent}" not found, inheritance not possible')
+        self._parent = parent
+        self.depth = found.depth + 1
+
+    def get_parent(self) -> str:
```

Comment:
That's not a property. That's a normal method. Should we adjust the docstring or has this method the wrong name? A property could be called like so: `obj.property` whilst this needs to be called like the following: `obj.method()`.

Reply:
fixed in https://github.com/cobbler/cobbler/pull/3266/commits/d8117a8cb4f51494abf7ebc60d0d9421704cf042